Feature PHPDoc check

kabalin opened the pull request:
This feature adds phpdoc check using [moodle-local_moodlecheck](https://github.com/moodlehq/moodle-local_moodlecheck) plugin (closing #10).

This command does not include command unittest though, as opposed to other commands, in order to be able to run phpdoc check as a part of unittest, a full Moodle installation is required (local_moodlecheck is using core moodle functions). If unittest is required, I guess I need to add an alternative to MoodleTestCase that would included cloning process instead of using fixture and use it for this command test.

kabalin committed:
Add Moodle PHPDoc checker.

This closes #10.

kabalin committed:
Refect changes in docs.

kabalin committed:
Mention MOODLE_REPO env variable in the docs.

kabalin committed:
Add phpdoc check to travis.

polothy commented on docs/TravisFileExplained.md:
```diff
@@ -41,8 +41,10 @@ php:
+# This line determines which version branch of Moodle to test against.
   - MOODLE_BRANCH=MOODLE_35_STABLE
+# Optionally, it is possible to specify a different Moodle repo to use:
+# - MOODLE_REPO=git://github.com/username/moodle.git
```
Please move these two lines up under line 44 - don't want to confuse them with the following block

kabalin replied:
Addressed in a6a0de7

polothy commented on src/Command/PHPDocCommand.php:
```diff
@@ -0,0 +1,74 @@
+        $process = $this->execute->passThroughProcess(
+            ProcessBuilder::create()
+                ->setPrefix('php')
+                ->add('local/moodlecheck/cli/moodlecheck.php')
```
I'm guessing this script doesn't take a list of files?  There is a built in way to ignore files (if you look at other commands like PHPLint).  Would be nice if this, at the very least, skipped vendors defined in `thirdpartylibs.xml` or whatever it is called.

kabalin replied:
Good point. it actually can take a list. I changed code to address this in 9dc2c9e

polothy commented on .travis.yml:
```diff
@@ -54,6 +54,7 @@ script:
   - moodle-plugin-ci savepoints
   - moodle-plugin-ci mustache
   - moodle-plugin-ci grunt
+  - moodle-plugin-ci phpdoc
```
Please add this same line to the `.travis.dist.yaml` file.

kabalin replied:
Addressed in a6a0de7

polothy reviewed (changes requested):
The build is failing due to code formatting I think.  Please run `make test` and it'll take care of it for you (commit the result).

kabalin committed:
Minor fixes following PR review.

kabalin committed:
Avoid running PHPDoc checker on excluded files.

Instead of directory, use MoodlePlugin::getFiles call to identify the
list of files required for testing and pass them to moodlecheck cli
script.

kabalin commented:
Hello @polothy 
Thanks for review! All listed issues are now addressed in two latest commits. Please let me know if further changes are required.

kabalin committed:
Add missing docblock for xmldb_local_travis_upgrade.

kabalin commented:
Heh, previous build apparently failed on phpdoc check 😃(https://travis-ci.org/blackboard-open-source/moodle-plugin-ci/jobs/481097455#L728) Fixed in b58488f.
